src/main/java/org/folio/oaipmh/processors/MarcWithHoldingsRequestHelper.java

```diff
@@ -677,45 +674,71 @@ private HttpClientRequest createEnrichInventoryClientRequest(HttpClient httpClie
       List<JsonObject> batch, boolean deletedRecordSupport) {
     final List<String> listOfIds = extractListOfIdsForSRSRequest(batch);
     logger.debug("Request to SRS, list id size: {}", listOfIds.size());
+    AtomicInteger attemptsCount = new AtomicInteger(REQUEST_ATTEMPTS);
     Promise<Map<String, JsonObject>> promise = Promise.promise();
+    doPostRequestToSrs(srsClient, deletedRecordSupport, listOfIds, attemptsCount, promise);
+    return promise.future();
+  }
+
+  private void doPostRequestToSrs(SourceStorageSourceRecordsClient srsClient, boolean deletedRecordSupport, List<String> listOfIds, AtomicInteger attemptsCount, Promise<Map<String, JsonObject>> promise) {
     try {
-      final Map<String, JsonObject> result = Maps.newHashMap();
-      srsClient.postSourceStorageSourceRecords("INSTANCE", deletedRecordSupport, listOfIds, srsResp -> srsResp.bodyHandler(bh -> {
-        if (srsResp.statusCode() != 200) {
-          String errorMsg = getErrorFromStorageMessage("source-record-storage", srsResp.request()
-            .absoluteURI(), srsResp.statusMessage());
-          srsClient.close();
-          handleException(promise, new IllegalStateException(errorMsg));
-          return;
-        }
-        try {
-          final Object o = bh.toJson();
-          if (o instanceof JsonObject) {
-            JsonObject entries = (JsonObject) o;
-            final JsonArray records = entries.getJsonArray("sourceRecords");
-            records.stream()
-              .filter(Objects::nonNull)
-              .map(JsonObject.class::cast)
-              .forEach(jo -> result.put(jo.getJsonObject("externalIdsHolder")
-                .getString(INSTANCE_ID_FIELD_NAME), jo));
+      srsClient.postSourceStorageSourceRecords("INSTANCE", deletedRecordSupport, listOfIds, srsResponse -> {
+        int statusCode = srsResponse.statusCode();
+        String statusMessage = srsResponse.statusMessage();
+        if (statusCode != 200) {
+          if(statusCode >= 500) {
+            String warnMessage = "Got error response form SRS, status code: " + statusCode + ", status message: {}" + statusMessage;
+            logger.warn(warnMessage);
+            if (attemptsCount.get() <= 0) {
+              String errorMessage = "SRS didn't respond with expected status code after " + REQUEST_ATTEMPTS + " attempts. Canceling further request processing.";
+              srsClient.close();
+              handleException(promise, new IllegalStateException(errorMessage));
+              return;
+            }
+            logger.warn("Trying to request SRS again, attempts left: {}", attemptsCount.decrementAndGet());
+            doPostRequestToSrs(srsClient, deletedRecordSupport, listOfIds, attemptsCount, promise);
```

Comment:
We probably want to have some retry backoff instead of just immediately trying again.  Ideally it would be configurable, but to reduce scope creep we can use a static constant.  It should be at least a second or two IMO

Reply:
Thank you for the suggestion. I added vertx timer here with a 1-second delay.